tests/unit/onyxTest.js

```diff
@@ -145,18 +147,88 @@ describe('Onyx', () => {
     it('should merge 2 arrays when it has no initial stored value for test key', () => {
         let testKeyValue;
         connectionID = Onyx.connect({
-            key: TEST_KEY,
+            key: ONYX_KEYS.TEST_KEY,
             initWithStoredValues: false,
             callback: (value) => {
                 testKeyValue = value;
             },
         });
 
-        Onyx.merge(TEST_KEY, ['test1']);
-        Onyx.merge(TEST_KEY, ['test2']);
+        Onyx.merge(ONYX_KEYS.TEST_KEY, ['test1']);
+        Onyx.merge(ONYX_KEYS.TEST_KEY, ['test2']);
         return waitForPromisesToResolve()
             .then(() => {
                 expect(testKeyValue).toEqual(['test1', 'test2']);
             });
     });
+
+    it('should properly set and merge when using mergeCollection', () => {
+        const valuesReceived = {};
+        let numberOfCallbacks = 0;
+
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_KEY,
+            initWithStoredValues: false,
+            callback: (data) => {
+                ++numberOfCallbacks;
```

Comment:
This is also fine but tossing out that `jest.fn()` can maybe can help here? We can look at the end to see how many calls were made and what each one was made with:

```js
const mockCallback = jest.fn()
connectionID = Onyx.connect({
            key: ONYX_KEYS.COLLECTION.TEST_KEY,
            initWithStoredValues: false,
            callback: mockCallback,
    });
    ...
    expect(mockCallback.mock.calls.length).toBe(7);
    expect(mockCallback.mock.calls[0][0]).toEqual({ID: 123, value: 'one'});
```

Reply:
I'll try that out

Reply:
oh damn, mock functions are pretty cool. Didn't know i can check number of times it was called with calls.length. Cool stuff. I do want to see if i can save the values somewhere to confirm the merge worked correctly than a set so I'll check out how to configure my mock function.

Reply:
yes! Can always save the value like you are doing and just call the mock function. So that example would become:

```js
    Onyx.connect({
            key: ONYX_KEYS.COLLECTION.TEST_KEY,
            initWithStoredValues: false,
            callback: (val) => {
                // save the value
               mockCallback();
            },
    });
```

Reply:
thats cool, i was actually thinking of setting the value in the mock function like so,
```
const mockCallback = jest.fn(data => valuesReceived[data.ID] = data.value);
```
That should be fine right? Or is it recommended to have the mock function empty?

Reply:
Nice! Can't think of any reason why we shouldn't do it as you've done.